src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -1043,6 +1043,23 @@ mgp_error mgp_map_insert(mgp_map *map, const char *key, mgp_value *value) {
   });
 }
 
+mgp_error mgp_map_update(mgp_map *map, const char *key, mgp_value *value) {
+  return WrapExceptions([&] {
+    auto emplace_result = map->items.emplace(key, *value);
+    if (!emplace_result.second) {
+      map->items.erase(emplace_result.first);
+      map->items.emplace(key, *value);
```

Comment:
I think it would be simpler to:
`emplace_result.first->second = *value`
but the = operator is deleted so that wasn't an option.